src/problem2.rb

```diff
@@ -0,0 +1,46 @@
+class Problem2
+  def calculateMaximum(values) #theta(n/2) -> O(n)
+    response = [nil,nil]
+    values.each_slice(2) do |first,second|
+      setMaxNumbers response,first,second
+    end
+    switchMaxValues response
+    return response
+  end
+
+  def setMaxNumbers(maxValues, firstValue, secondValue)
+    if maxValues[0] == nil && maxValues[1] == nil
+      maxValues[0] = firstValue
+      maxValues[1] = secondValue.nil? ? firstValue : secondValue
+      return
+    end
+    compareValues maxValues,firstValue
+    if secondValue != nil
+      compareValues maxValues,secondValue
```

Comment:
Can you find the cause for calling `compareValues` twice and the `nil` checks ?

Reply:
It's because i take numbers two by two, and the null check is required when we encounter arrays of odd length. This whole solution could have been much simpler if i would have taken numbers one by one.